messagingpush/src/main/java/io/customer/messagingpush/activity/NotificationClickReceiverActivity.kt

```diff
@@ -0,0 +1,135 @@
+package io.customer.messagingpush.activity
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import androidx.core.app.TaskStackBuilder
+import io.customer.messagingpush.MessagingPushModuleConfig
+import io.customer.messagingpush.config.NotificationClickBehavior
+import io.customer.messagingpush.data.model.CustomerIOParsedPushPayload
+import io.customer.messagingpush.di.deepLinkUtil
+import io.customer.messagingpush.di.moduleConfig
+import io.customer.messagingpush.extensions.parcelable
+import io.customer.messagingpush.util.DeepLinkUtil
+import io.customer.sdk.CustomerIO
+import io.customer.sdk.CustomerIOShared
+import io.customer.sdk.data.request.MetricEvent
+import io.customer.sdk.extensions.takeIfNotBlank
+import io.customer.sdk.tracking.TrackableScreen
+import io.customer.sdk.util.Logger
+
+/**
+ * Activity to handle notification click events.
+ *
+ * This activity is launched when a notification is clicked. It tracks opened
+ * metrics, handles the deep link and opens the desired activity in the host app.
+ */
+class NotificationClickReceiverActivity : Activity(), TrackableScreen {
+    val logger: Logger by lazy { CustomerIOShared.instance().diStaticGraph.logger }
+
+    override fun getScreenName(): String? {
+        // Return null to prevent this screen from being tracked
+        return null
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        handleIntent(data = intent)
+    }
+
+    override fun onNewIntent(intent: Intent?) {
+        super.onNewIntent(intent)
+        handleIntent(data = intent)
+    }
+
+    private fun handleIntent(data: Intent?) {
+        kotlin.runCatching {
+            val payload: CustomerIOParsedPushPayload? =
+                data?.extras?.parcelable(NOTIFICATION_PAYLOAD_EXTRA)
+            if (payload == null) {
+                logger.error("Payload is null, cannot handle notification intent")
+            } else {
+                processNotificationIntent(payload = payload)
+            }
+        }.onFailure { ex ->
+            logger.error("Failed to process notification intent: ${ex.message}")
+        }
+        finish()
+    }
+
+    private fun processNotificationIntent(payload: CustomerIOParsedPushPayload) {
```

Comment:
can we further break down the logic here so it gets a little less complicated and more maintable? maybe a separate method of `handling deep linking` ?

Reply:
Updated